addons/marketing_campaign/models/marketing_campaign_segment.py

```diff
@@ -0,0 +1,152 @@
+                                   "If no filter is set, all records are selected without filtering. "
+                                   "The synchronization mode may also add a criterion to the filter.")
+    sync_last_date = fields.Datetime(
+        string='Last Synchronization', help="Date on which this segment was synchronized last time (automatically or manually)")
+    sync_mode = fields.Selection([('create_date', 'Only records created after last sync'),
+                                  ('write_date',
+                                   'Only records modified after last sync (no duplicates)'),
+                                  ('all', 'All records (no duplicates)')],
+                                 string='Synchronization mode', default='create_date', help="Determines an additional criterion to add to the filter when selecting new records to inject in the campaign. "
+                                 '"No duplicates" prevents selecting records which have already entered the campaign previously.'
+                                 'If the campaign has a "unique field" set, "no duplicates" will also prevent selecting records which have '
+                                 'the same value for the unique field as other records that already entered the campaign.')
+    state = fields.Selection([('draft', 'New'),
+                              ('cancelled', 'Cancelled'),
+                              ('running', 'Running'),
+                              ('done', 'Done')],
+                             'Status', copy=False, default='draft')
+    date_run = fields.Datetime(
+        string='Launch Date', help="Initial start date of this segment.")
+    date_done = fields.Datetime(
+        string='End Date', help="Date this segment was last closed or cancelled.")
+    date_next_sync = fields.Datetime(compute='_get_next_sync', string='Next Synchronization',
+                                     help="Next time the synchronization job is scheduled to run automatically")
+
+    @api.one
+    @api.constrains('ir_filter_id', 'campaign_id')
+    def _check_model(self):
+        if not self.ir_filter_id:
+            return True
+        if self.campaign_id.object_id.model != self.ir_filter_id.model_id:
+            raise UserError(
+                _('Model of filter must be same as resource model of Campaign.'))
+        return True
+
+    @api.onchange('campaign_id')
+    def onchange_campaign_id(self):
+        res = {'domain': {'ir_filter_id': []}}
+        if self.campaign_id:
+            campaign = self.campaign_id
+            if campaign.object_id:
+                mod_name = campaign.object_id.model
+                res['domain'] = {'ir_filter_id': [('model_id', '=', mod_name)]}
+        else:
+            self.ir_filter_id = False
+        return res
+
+    @api.multi
+    def state_running_set(self):
+        vals = {'state': 'running'}
+        if not self.date_run:
+            vals['date_run'] = fields.Date.context_today(self)
```

Comment:
I think, it's better to use fields.Date.today instead of fields.Date.context_today(self)